newIDE/electron-app/app/main.js

```diff
@@ -174,81 +182,58 @@ app.on('ready', function() {
     });
   });
 
-  ipcMain.on('piskel-open-then-load-animation', (event, externalEditorData) => {
-    loadModalWindow({
+  // Piskel image editor
+  ipcMain.handle('piskel-load', (event, externalEditorInput) => {
```

Comment:
what's the difference between `icpMain.on()` and `icpMain.handle()` ?

Reply:
`ipcMain.handle` works with promise, so you can do on the renderer side `await ipcRenderer.invoke('your-event', ...)`.
While ipcMain.on works with event listeners, which works too but is less convenient.

`ipcMain.handle` is fairly new in Electron, hence why it's not used everywhere. But it must be preferred now!

Reply:
See https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrendererinvokechannel-args